cm_sdk/models/__init__.py

```diff
@@ -0,0 +1,251 @@
+        if isinstance(o, CloudManagerEnum):
+            return o.to_json()
+        elif hasattr(o, "state"):
+            # converts class parameter names from snake case to camel case expected
+            # by the API and prunes nulls
+            return dict((StringUtil.to_camel_case(k), v) for k, v in o.state.iteritems() if v)
+        else:
+            return None
+
+
+class CloudManagerBase(object):
+
+    """
+    Base class for python representations of CloudManager API objects that provides
+    utility methods for static initialization, factors out common attributes and
+    provides common API object equivalence checks.
+    """
+    # All API objects have these attributes
+    COMMON_API_ATTRIBUTES = ['id', 'group_id', 'links', 'created']
+    # The following attributes are ignored for determine whether instances are equivalent.
+    # This is useful when determining if a non-persitant entity duplicates one that was
+    # created previously.
+    EQUIVALENCE_IGNORED_ATTRIBUTES = ['id', 'group_id', 'links', 'created', 'updated']
+
+    def __init__(self, my_api_attributes):
+
+        all_api_attributes = list(self.COMMON_API_ATTRIBUTES)
+        all_api_attributes.extend(my_api_attributes)
+        vars(self).update(all_api_attributes=all_api_attributes, state={})
+
+    def equivalent(self, other):
+        """
+        Compares two API Objects, but ignoring specified attributes.
+        This allows determining whether an object created locally
+        is equivalent to one that was create previously.
+        """
+
+        _self = json.loads(self.to_json())
+        _other = json.loads(other.to_json())
+        filtered_self = self.remove_keys(_self, CloudManagerBase.EQUIVALENCE_IGNORED_ATTRIBUTES)
+        filtered_other = self.remove_keys(_other, CloudManagerBase.EQUIVALENCE_IGNORED_ATTRIBUTES)
+
+        return filtered_self == filtered_other
+
+        # return self._compare_state(filtered_self, filtered_other)
+
+    @staticmethod
+    def remove_keys(obj, to_remove):
+        """
+        Removes specified keys from a nested dict.  Used to determined the deep equivalence of API objects.  Keys, like
+        ID, created, updated, that are set on the server side do not materially influence the equivalence of objects
+        like AlertConfigs or Backups.
+        """
+        if isinstance(obj, dict):
+            obj = {
+                key: CloudManagerBase.remove_keys(value, to_remove)
+                for key, value in obj.iteritems()
+                if key not in to_remove}
+        elif isinstance(obj, list):
+            obj = [CloudManagerBase.remove_keys(item, to_remove)
+                   for item in obj
+                   if item not in to_remove]
+        return obj
+
+    @staticmethod
+    def get_class_from_name(kls):
+        parts = kls.split('.')
+        module = ".".join(parts[:-1])
+        m = __import__(module)
+        for comp in parts[1:]:
+            m = getattr(m, comp)
```

Comment:
This can throw an exception while you're digging into an object.  It would be useful to know the fully qualified path(`kls`) that you were trying to resolve when that happens.  You should add a try/catch around this.